vwr/driver.py

```
@@ -59,18 +55,18 @@ def _reconnect(self):
         if self.reconnect_trials < 10:
             try:
                 self._connect()
-                self.reconnect_loop.stop()
+                self.get_setpoint()
+                self.reconnect_loop = Timer(self.initialDelay, self._reconnect)
                 self.reconnect_trials = 0
             except:
```

I'd specify whatever exception type happens when disconnected. This ensures that this loop doesn't accidentally silence other exception types.

---

I'm 90% sure that it's a `socket.timeout` exception but I'll need to go downstairs and reproduce the error to make sure. Should I do that or do you also think that it's the `socket.timeout` exception